lib/scanner/blocks.go

```diff
@@ -215,3 +229,7 @@ func (noopHash) Size() int                 { return 0 }
 func (noopHash) Reset()                    {}
 func (noopHash) Sum([]byte) []byte         { return nil }
 func (noopHash) Write([]byte) (int, error) { return 0, nil }
+
+type NoopCounter struct{}
```

Comment:
can be private (lower case name)

Reply:
no, replaced `nil` counters with this `NoopCounter `

Reply:
so we push nil to the boundaries

Reply:
Still can be private.

Reply:
I don't really see the point having a type for just one nil check?

Reply:
Actually, ignore my comment above, but Audrius' one about private/not exported still is unaddressed.